opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -259,14 +259,15 @@ def end(self, end_time: Optional[int] = None) -> None:
                 raise RuntimeError("Calling end() on a not started span.")
```

Comment:
a random aside, but there's a lot of stuff in this lock that we can probably move out (e.g. the check to see if we're recording traces).

Not sure what sort of contention we'll see here, but keeping the locks tight is a good thing.